### PR TITLE
Support templating generated secret

### PR DESCRIPTION
**Pull request by allanger**

### CODE REVIEW

File: controllers/database_helper.go [hunk 1]
```diff
@@ -39,6 +40,19 @@ type ConnectionStringFields struct {
...
+const (
+	fieldPostgresDB        = "POSTGRES_DB"
+	fieldPostgresUser      = "POSTGRES_USER"
+	fieldPostgressPassword = "POSTGRES_PASSWORD"
+	fieldMysqlDB           = "DB"
+	fieldMysqlUser         = "USER"
+	fieldMysqlPassword     = "PASSWORD"
+)
+
+func getUntemplatedSecrets() []string {
```
Comment (dabde) [inline controllers/database_helper.go:52]: don't understand this name.

Comment (dabde) [reply controllers/database_helper.go:52]: would change this to getBlockedTempatedKeys.

File: api/v1alpha1/database_types.go [hunk 2]
```diff
@@ -35,8 +35,9 @@ type DatabaseSpec struct {
 	// These keywords can be used: Protocol, DatabaseHost, DatabasePort, UserName, Password, DatabaseName.
 	// Default template looks like this:
 	// "{{ .Protocol }}://{{ .UserName }}:{{ .Password }}@{{ .DatabaseHost }}:{{ .DatabasePort }}/{{ .DatabaseName }}"
-	ConnectionStringTemplate string   `json:"connectionStringTemplate,omitempty"`
-	Postgres                 Postgres `json:"postgres,omitempty"`
+	ConnectionStringTemplate string            `json:"connectionStringTemplate,omitempty"`
+	SecretsTemplates         map[string]string `json:"secretsTemplates,omitempty"`
```
Comment (dabde) [inline api/v1alpha1/database_types.go:39]: think we should also write a readme for this feature, or?

Comment (allanger) [reply api/v1alpha1/database_types.go:39]: I've updated docs here: https://github.com/kloeckner-i/db-operator/blob/155-secrets-templating/docs/creatingdatabases.md

**Review by dabde — APPROVED** [review]

**Pull request merged**